### PR TITLE
unreal_asset: split into multiple crates

### PR DESCRIPTION
**Pull request by localcc**
This is done to improve lint performance

### CODE REVIEW

File: unreal_asset/src/lib.rs [hunk 1]
```diff
@@ -38,123 +38,56 @@
 //! println!("{:#?}", asset);
 //! ```
 
-use std::fmt::{Debug, Formatter};
-use std::io::{Read, Seek, SeekFrom, Write};
-use std::mem::size_of;
+use std::{
```
Comment (konsti219) [inline unreal_asset/src/lib.rs:41]: wrong import style

File: unreal_asset/src/lib.rs [hunk 2]
```diff
@@ -38,123 +38,56 @@
...
-use containers::{chain::Chain, indexed_map::IndexedMap, shared_resource::SharedResource};
-use custom_version::{CustomVersion, CustomVersionTrait};
-use engine_version::EngineVersion;
-use enums::ECustomVersionSerializationFormat;
-use error::Error;
-use exports::{
-    base_export::BaseExport, class_export::ClassExport, Export, ExportBaseTrait, ExportNormalTrait,
-    ExportTrait,
+
+use crate::{
```
Comment (konsti219) [inline unreal_asset/src/lib.rs:59]: imports

File: unreal_asset/src/reader/asset_archive_writer.rs [hunk 3]
```diff
@@ -1,20 +1,20 @@
 //! Archive that can be used to write an asset
 
 use crate::{
```
Comment (konsti219) [inline unreal_asset/src/reader/asset_archive_writer.rs:3]: imports

File: unreal_asset/src/reader/mod.rs [hunk 4]
```diff
@@ -1,7 +1,3 @@
-//! Archive readers/writers
-pub mod archive_reader;
-pub mod archive_trait;
-pub mod archive_writer;
+//! Unreal asset asset-specific readers
+
 pub mod asset_archive_writer;
```
Comment (konsti219) [inline unreal_asset/src/reader/mod.rs]: consider removing the folder/module?

File: unreal_asset/unreal_asset_base/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,45 @@
+[package]
+name = "unreal_asset_base"
+version.workspace = true
+authors.workspace = true
+description = "Unreal Engine 4/5 game asset serialization library."
```
Comment (konsti219) [inline unreal_asset/unreal_asset_base/Cargo.toml:5]: We need descriptions for the structure of the crates somewhere.

File: unreal_asset/unreal_asset_base/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+zstd = "0.12.4"
+flate2 = "1.0.25"
+lz4_flex = { version = "0.11.1", features = [
+    "safe-decode",
+    "safe-encode",
+    "std",
+], default-features = false }
+
+# oodle compression
+libloading = "0.8.0"
```
Comment (konsti219) [inline unreal_asset/unreal_asset_base/Cargo.toml:34]: huh?

File: unreal_asset/unreal_asset_exports/Cargo.toml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+[package]
+name = "unreal_asset_exports"
+version.workspace = true
+authors.workspace = true
+description = "Unreal Engine 4/5 game asset serialization library."
+license.workspace = true
+homepage.workspace = true
+documentation = "https://docs.rs/unreal_asset_exports"
+edition = "2021"
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (konsti219) [inline unreal_asset/unreal_asset_exports/Cargo.toml:10]: how often is this comment found around the project?

File: unreal_asset/unreal_asset_exports/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,7 @@
+//! Unreal asset exports
+
+use unreal_asset_base::*;
```
Comment (konsti219) [inline unreal_asset/unreal_asset_exports/src/lib.rs:3]: Why is exports a module? Can't the contents be lifted to the crate root? These imports are useless?

File: unreal_asset/unreal_asset_kismet/src/lib.rs [hunk 9]
```diff
@@ -0,0 +1,5 @@
+//! Unreal asset kismet implementations
+
+use unreal_asset_base::*;
```
Comment (konsti219) [inline unreal_asset/unreal_asset_kismet/src/lib.rs:3]: why this import here?

File: unreal_asset/unreal_asset_properties/src/lib.rs [hunk 10]
```diff
@@ -0,0 +1,6 @@
+#![deny(missing_docs)]
+//! Unreal asset properties
+
+pub use unreal_asset_base::*;
```
Comment (konsti219) [inline unreal_asset/unreal_asset_properties/src/lib.rs:4]: ?

File: unreal_asset/unreal_asset_properties/src/properties/movies/movie_scene_eval_template_ptr_property.rs [hunk 11]
```diff
@@ -2,7 +2,16 @@
 
 use crate::properties::property_prelude::*;
 
-use crate::properties::str_property::StrProperty;
+use crate::{
```
Comment (konsti219) [inline unreal_asset/unreal_asset_properties/src/properties/movies/movie_scene_eval_template_ptr_property.rs:5]: imports

**Review by konsti219 — CHANGES_REQUESTED** [review]

File: unreal_asset/unreal_asset_exports/src/lib.rs (hunk 8 above)
Comment (localcc) [reply unreal_asset/unreal_asset_exports/src/lib.rs:3]: mostly better lint performance, these imports are required so imports don't have to be changed in every source file

**Review by konsti219 — APPROVED** [review]

**Pull request merged**